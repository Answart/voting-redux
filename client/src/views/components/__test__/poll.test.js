import React from 'react';
import * as d3 from 'd3';
import Poll from '../poll';
import {
  muiMounter, mockPoll
} from '../../../utils/__test__';


describe('<Poll />', () => {
  let wrapper ;
  beforeAll(async () => {
    const props = {
      votes: mockPoll.votes,
      choices: mockPoll.choices
    };
    wrapper = muiMounter(<Poll {...props} />);
    await wrapper.find(Poll).instance().componentDidMount();
  });

  it('renders properly', async () => {
    const page = wrapper.find('Poll');
    expect(page).toHaveLength(1);
    expect(Object.keys(page.props()).length).toBe(2);
    expect(page.prop('votes')).toBe(10);
    expect(page.prop('choices')).toEqual([
      { id: 0, label: 'Springer Spaniel', vote: 4 },
      { id: 1, label: 'Cocker Spaniel', vote: 6 }
    ]);
    expect(page.find('div#chart')).toHaveLength(1);
    expect(typeof page.instance().renderChart).toBe('function');
    expect(mountToJson(page)).toMatchSnapshot();
  });

  it('renders d3 chart properly on mount (renderChart())', async () => {
    let chart = d3.select('#chart');
    let choices = chart.selectAll('div.choice').nodes();
    let labels = chart.selectAll('div.poll-chart-label').nodes();
    let bars = chart.selectAll('.poll-chart-bar').nodes();
    let votes = chart.selectAll('.poll-chart-bar').nodes();

    expect(d3.select(choices[0]).style('height')).toBe('40px');
    expect(d3.select(choices[1]).style('height')).toBe('40px');

    expect(d3.select(labels[0]).text()).toBe('Springer Spaniel');
    expect(d3.select(labels[1]).text()).toBe('Cocker Spaniel');

    expect(d3.select(bars[0]).style('height')).toBe('20px');
    expect(d3.select(bars[0]).style('width')).toBe('0px');
    expect(d3.select(bars[1]).style('height')).toBe('20px');
    expect(d3.select(bars[1]).style('width')).toBe('0px');

    expect(d3.select(votes[0]).text()).toBe('4');
    expect(d3.select(votes[1]).text()).toBe('6');
  });
});
