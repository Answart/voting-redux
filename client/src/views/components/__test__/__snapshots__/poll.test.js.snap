// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Poll /> renders properly 1`] = `
<Poll
  choices={
    Array [
      Object {
        "id": 0,
        "label": "Springer Spaniel",
        "vote": 4,
      },
      Object {
        "id": 1,
        "label": "Cocker Spaniel",
        "vote": 6,
      },
    ]
  }
  votes={10}
>
  <WithStyles(Paper)
    className="poll-chart-paper"
  >
    <Paper
      className="poll-chart-paper"
      classes={
        Object {
          "elevation0": "MuiPaper-elevation0-3",
          "elevation1": "MuiPaper-elevation1-4",
          "elevation10": "MuiPaper-elevation10-13",
          "elevation11": "MuiPaper-elevation11-14",
          "elevation12": "MuiPaper-elevation12-15",
          "elevation13": "MuiPaper-elevation13-16",
          "elevation14": "MuiPaper-elevation14-17",
          "elevation15": "MuiPaper-elevation15-18",
          "elevation16": "MuiPaper-elevation16-19",
          "elevation17": "MuiPaper-elevation17-20",
          "elevation18": "MuiPaper-elevation18-21",
          "elevation19": "MuiPaper-elevation19-22",
          "elevation2": "MuiPaper-elevation2-5",
          "elevation20": "MuiPaper-elevation20-23",
          "elevation21": "MuiPaper-elevation21-24",
          "elevation22": "MuiPaper-elevation22-25",
          "elevation23": "MuiPaper-elevation23-26",
          "elevation24": "MuiPaper-elevation24-27",
          "elevation3": "MuiPaper-elevation3-6",
          "elevation4": "MuiPaper-elevation4-7",
          "elevation5": "MuiPaper-elevation5-8",
          "elevation6": "MuiPaper-elevation6-9",
          "elevation7": "MuiPaper-elevation7-10",
          "elevation8": "MuiPaper-elevation8-11",
          "elevation9": "MuiPaper-elevation9-12",
          "root": "MuiPaper-root-1",
          "rounded": "MuiPaper-rounded-2",
        }
      }
      component="div"
      elevation={2}
      square={false}
    >
      <div
        className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 poll-chart-paper"
      >
        <div
          className="poll-chart"
          id="chart"
        />
      </div>
    </Paper>
  </WithStyles(Paper)>
</Poll>
`;
